test/app/components/content_areas_component.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class ContentAreasComponent < ActionView::Component::Base
+  validates :body, :title, :footer, presence: true
+
+  with_content_areas :title, :body, :footer
+
+  def initialize(title: nil, footer: nil)
+    @title = title
```

Comment:
I think it's a little awkward to require this ivar setting / method arguments to make this interface work. 

IMO we should remove this for this first PR and see if we find ourselves needing this interface in the future, and if so, what a simpler implementation might look like at the component usage level.

Reply:
We aren't writing any code to make this work. We'd have to write code to check that content_areas don't have the same name as constructor params. I think we should just leave the option to use both methods. It will be clearer with better examples.

Reply:
Added some examples to the Readme. I can see places where various combinations are valid/powerful. They might not be the most common cases but IMO none of them are confusing or unexpected. To prevent some or all of those combinations we'd have to write more code. Specifically to prevent content areas having names that are named parameters of the Component constructor. Going out of our way to write more code to restrict flexibility doesn't feel like the right tradeoff to me.